tmt/utils.py

```diff
@@ -2632,18 +2634,47 @@ def remove_color(text: str) -> str:
     return re.sub(r'\x1B(?:[@-Z\\-_]|\[[0-?]*[ -/]*[@-~])', '', text)
 
 
-def default_branch(repository: Path, remote: str = 'origin') -> Optional[str]:
+def default_branch(repository: Path, logger: tmt.log.Logger,
+                   remote: str = 'origin') -> Optional[str]:
     """ Detect default branch from given local git repository """
+    # Make sure '.git' (which is a file or a directory) is present
+    dot_git = repository / '.git'
+    if not dot_git.exists():
+        return None
+
+    if not dot_git.is_file() and not dot_git.is_dir():
+        return None
+
+    work_repo = repository
+    if dot_git.is_file():
+        # Get the parent path of the worktree and the .git file looks like:
+        #     gitdir: /tmp/foo/REPRO/.git/worktrees/TREE
+        command = Command("git", "rev-parse", "--path-format=absolute", "--git-common-dir")
+        try:
+            result = run_command(command=command, cwd=Path(repository), logger=logger)
```

Comment:
`repository` is already `Path`, no need to convert it again.

Reply:
Fixed in 27e8add5.